### PR TITLE
Add transaction metadata. Make some operation types more explicit.

### PR DESCRIPTION
**Pull request by andreibancioiu**
- Add [transaction metadata](https://www.rosetta-api.org/docs/models/Transaction.html): 

```
{
	"timestamp":         ...,
	"value":             ...,
	"nonce":             ...,
	"typeOnSource":      ...,
	"typeOnDestination": ...,
	"epoch":             ...,
	"sender":            ...,
	"receiver":          ...,
	"sourceShard":       ...,
	"destinationShard":  ...,
	"miniblock":         ...,
	"miniblockType":     ...,
	...
}
```

 - Add new operation types, to better differentiate operations: `FeeRefundAsSmartContractResult`, `DeveloperRewardsAsSmartContractResult`.

- Improve error handling for `network/status`, when `highestFinalNonce == 0` (node didn't start syncing).

### CODE REVIEW

File: server/services/operations.go [hunk 1]
```diff
@@ -3,13 +3,15 @@ package services
...
 
 const (
-	opGenesisBalanceMovement = "GenesisBalanceMovement"
-	opTransfer               = "Transfer"
-	opFee                    = "Fee"
-	opReward                 = "Reward"
-	opScResult               = "SmartContractResult"
-	opFeeOfInvalidTx         = "FeeOfInvalidTransaction"
-	opFeeRefund              = "FeeRefund"
+	opGenesisBalanceMovement     = "GenesisBalanceMovement"
```
Comment (andreibancioiu) [inline server/services/operations.go:6]: Please confirm the operation types are sufficient & consistent.

File: server/services/transactionMetadata.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		metadata["receiverUsername"] = string(tx.ReceiverUsername)
+	}
+	if len(tx.OriginalTransactionHash) > 0 {
+		metadata["originalTransaction"] = tx.OriginalTransactionHash
+	}
+	if len(tx.PreviousTransactionHash) > 0 {
+		metadata["previousTransaction"] = tx.PreviousTransactionHash
+	}
+	if len(tx.Data) > 0 {
+		metadata["data"] = tx.Data
```
Comment (andreibancioiu) [inline server/services/transactionMetadata.go:37]: Will get serialized as base64 in the response JSON.

File: server/provider/nodeStatus_test.go [hunk 3]
```diff
@@ -122,6 +136,58 @@ func TestNetworkProvider_GetNodeStatusWithError(t *testing.T) {
...
+	args := createDefaultArgsNewNetworkProvider()
+	args.ObserverFacade = observerFacade
+	args.FirstHistoricalEpoch = 2
+	args.NumHistoricalEpochs = 8
+
+	provider, err := NewNetworkProvider(args)
+	require.Nil(t, err)
+	require.NotNil(t, provider)
+
+	t.Run("when HighestFinalNonce <= 2 (node didn't start syncing)", func(t *testing.T) {
```
Comment (bogdan-rosianu) [inline server/provider/nodeStatus_test.go:150]: can the tests be run in parallel?

File: server/services/transactionMetadata.go (hunk 2 above)
Comment (bogdan-rosianu) [reply server/services/transactionMetadata.go:37]: :+1:

File: server/services/transactionMetadata.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		"epoch":             tx.Epoch,
+		"sender":            tx.Sender,
+		"receiver":          tx.Receiver,
+		"sourceShard":       tx.SourceShard,
+		"destinationShard":  tx.DestinationShard,
+		"miniblock":         tx.MiniBlockHash,
+		"miniblockType":     tx.MiniBlockType,
+	}
+
+	if len(tx.OriginalSender) > 0 {
```
Comment (iulianpascalau) [inline server/services/transactionMetadata.go:21]: this is the "omit empty" implementation :D

**Review by iulianpascalau — COMMENTED** [review]
Only what @bogdan-rosianu said

**Review by iulianpascalau — APPROVED** [review]

File: server/provider/nodeStatus_test.go (hunk 3 above)
Comment (andreibancioiu) [reply server/provider/nodeStatus_test.go:156]: Added `t.Parallel()`.

**Review by bogdan-rosianu — APPROVED** [review]

File: server/services/transactionMetadata.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+func extractTransactionMetadata(tx *transaction.ApiTransactionResult) objectsMap {
+	metadata := objectsMap{
+		"timestamp":         tx.Timestamp,
+		"value":             tx.Value,
+		"nonce":             tx.Nonce,
+		"typeOnSource":      tx.ProcessingTypeOnSource,
+		"typeOnDestination": tx.ProcessingTypeOnDestination,
+		"epoch":             tx.Epoch,
+		"sender":            tx.Sender,
+		"receiver":          tx.Receiver,
```
Comment (AdoAdoAdo) [inline server/services/transactionMetadata.go:14]: if it is an esdt transfer, the receiver is not always in the tx.Receiver field

Comment (andreibancioiu) [reply server/services/transactionMetadata.go:14]: Thanks! We will handle that one here: https://github.com/ElrondNetwork/rosetta/pull/43

**Pull request merged**